### ISSUE TITLE
Android-bug-Crash/doesn't work offline 

### ISSUE THREAD

**Issue by avichou** [omnivore-app/omnivore#3017]
Putting the phone in airplane or simply not having connection will lead the app to ...simply not work. As soon as you try to open an article it just gives a blank image 

Android 13 , oneui5, Samsung s23 

Side question but how many articles get downloaded for offline use ? Is there a limit ? Like 1000 last articles of pocket app 

![Screenshot_20231025_223820_Omnivore.jpg](https://github.com/omnivore-app/omnivore/assets/86660671/dc0136a5-356a-4df1-9762-f80e394d1613)

![Screenshot_20231025_223824_Omnivore.jpg](https://github.com/omnivore-app/omnivore/assets/86660671/a3f29a4e-c21e-4770-8788-22185227b87c)

**Comment by kbdragon** [comment]
Yeah, this is not offline support, sadly. 
Any plans to add true offline support?

**Comment by cl93a** [comment]
@tierced , does the app load on your Pixel 6? On mine, it crashes whilst opening, every single time.

**Comment by tierced** [comment]
The app does load. That hasn't been an issue.

Mar 21, 2024 18:36:53 cl93a - notifications at github.com ***@***.***>:

**Comment by Marxsal** [comment]
I'm kind of surprised this thread is being ignored.

At a minimum, shouldn't the promise "Offline mode" be removed from the documentation? Or qualified, "Works on iPhone only" ?

**Comment by jacksonh** [comment]
> I'm kind of surprised this thread is being ignored.
> 
> At a minimum, shouldn't the promise "Offline mode" be removed from the documentation? Or qualified, "Works on iPhone only" ?

Hi Marxsal, can i ask which device you are having issues with offline on?

**Comment by kbdragon** [comment]
@jacksonh how big is the article cache on android? 
I just tried putting my Samsung a54 in airplane mode and opening the app and got a blank screen with only the top menu.
If I try to open the app, put the phone in airplane mode, then I have a few articles appear, but I can only open the top three.
Can we increase the cache to 10? Arbitrarily?
I wouldn't mind the app taking 1gb of space if it meant that I'd have my reading with me on a plane. 
Also, can we do a background sync to load new articles/newsletters without opening the app?

**Comment by jacksonh** [comment]
@kbdragon it should download all items for the search but on Android right now its doing that separately from fetching the article list, so there's a daily. I'm trying to update it to act more like iOS.

When you open in airplane mode the downloaded articles are disappearing?

**Comment by kbdragon** [comment]
@jacksonh I just checked, and yes, downloaded articles are disappearing, but not consistently. 
I connected to the network, opened omnivore, loaded an article, exited the app (but didn't close it from ram), turned on airplane mode, reopened the app and tried to load the same article - nothing. But, on the same session, I tried loading a previously read article, and that loaded just fine

**Comment by jacksonh** [comment]
> @jacksonh I just checked, and yes, downloaded articles are disappearing, but not consistently. I connected to the network, opened omnivore, loaded an article, exited the app (but didn't close it from ram), turned on airplane mode, reopened the app and tried to load the same article - nothing. But, on the same session, I tried loading a previously read article, and that loaded just fine

Do they disappear from your library or just have no content?

**Comment by Marxsal** [comment]
I'm using Fire 8, which corresponds to Android 11. And testing on Moto G6, Android 8.

Both systems exhibit problems.

What sometimes works is to not browse any articles. Turn off internet. Clear the cache. Turn the internet on. Allow the library to reload. Then turn the internet off again and try to browse.

Following this process I've gotten as many as 30 articles. But sometimes fewer articles. And the articles become inaccessible over time.

My feeling is that Omnivore is failing to check the internet status before loading articles. When you first load an article while connected to the internet there's a momentary blank screen. That screen stays around when you're browsing offline. In fact, the "back" button is still there (upper left hand corner), but it is invisible (white-on-white). So it seems like it is attempting to fetch remote articles rather than using it's own local copy.

There are other problems too. If I apply tags to an article, even if online, the tags don't show up until I've exited the app completely and gone back in.  I'm guessing it only applies the tags when fetching them from the remote site.

**Comment by kquinsland** [comment]
I noticed a slightly different set of issues w/ the Android app. This issue appears to be the only / major "android, offline doesn't work" ticket. If I should create a new ticket and link it here, let me know.

---

Most things were available to read offline but it seems hit/miss which highlights and notes made while offline actually persisted. 

Everything that I archived while offline showed back up once I regained WAN connection which was an "interesting" surprise; might be the same underlying cause as https://github.com/omnivore-app/omnivore/issues/2888 but unsure.
